fuse_models/test/test_unicycle_2d.cpp

```diff
@@ -291,8 +316,148 @@ TEST(Unicycle2D, UpdateStateHistoryEstimates)
+    auto velocity_linear = std::static_pointer_cast<fuse_variables::VelocityLinear2DStamped>(variables.at(7));
+    auto velocity_yaw = std::static_pointer_cast<fuse_variables::VelocityAngular2DStamped>(variables.at(8));
+    auto acceleration_linear = std::static_pointer_cast<fuse_variables::AccelerationLinear2DStamped>(variables.at(9));
+
+    auto expected_pose = tf2_2d::Transform(1.643536632, 3.498058629, 1.2);
+    auto expected_linear_velocity = tf2_2d::Vector2(2.0, 0.0);
+    auto expected_yaw_velocity = 0.0;
+    auto expected_linear_acceleration = tf2_2d::Vector2(1.0, 0.0);
+
+    EXPECT_NEAR(position->x(), expected_pose.x(), 1.0e-9);
+    EXPECT_NEAR(position->y(), expected_pose.y(), 1.0e-9);
+    EXPECT_NEAR(orientation->yaw(), expected_pose.yaw(), 1.0e-9);
+
+    EXPECT_NEAR(velocity_linear->x(), expected_linear_velocity.x(), 1.0e-9);
+    EXPECT_NEAR(velocity_linear->y(), expected_linear_velocity.y(), 1.0e-9);
+
+    EXPECT_NEAR(velocity_yaw->yaw(), expected_yaw_velocity, 1.0e-9);
+
+    EXPECT_NEAR(acceleration_linear->x(), expected_linear_acceleration.x(), 1.0e-9);
+    EXPECT_NEAR(acceleration_linear->y(), expected_linear_acceleration.y(), 1.0e-9);
+  }
+
+  // Now check the created constraint
+  {
+    auto new_constraint =
+      std::static_pointer_cast<fuse_models::Unicycle2DStateKinematicConstraint>(constraints.back());
+
+    // As based on the yaw angle we are mainly moving upwards and the process noise covariance
+    // is defined in the robot frame (original: cov_xx=1.0, cov_yy=2.0) it is expected that the final
+    // process noise covariance in the world frame (so the frame where optimization happens) is higher
+    // at cov_xx than in cov_yy.
+    // Additionally, cov_xy and cov_yx are also no more zero (so no more independent between x and y) because
+    // again it is moving simultaneously in the x- and y-direction.
+    fuse_core::Matrix8d expected_cov;
+    expected_cov <<  1.868696858, -0.33773159,        0,       0,       0,       0,       0,       0,
+                    -0.33773159,   1.131303142,       0,       0,       0,       0,       0,       0,
+                              0,             0,       3,       0,       0,       0,       0,       0,
+                              0,             0,       0,       4,       0,       0,       0,       0,
+                              0,             0,       0,       0,       5,       0,       0,       0,
+                              0,             0,       0,       0,       0,       6,       0,       0,
+                              0,             0,       0,       0,       0,       0,       7,       0,
+                              0,             0,       0,       0,       0,       0,       0,       8;
+
+    EXPECT_MATRIX_NEAR(expected_cov, new_constraint->covariance(), 1.0e-9);
+  }
+}
+
 int main(int argc, char **argv)
 {
   testing::InitGoogleTest(&argc, argv);
-  return RUN_ALL_TESTS();
+  ros::init(argc, argv, "unicycle_2d_test");
```

Comment:
Why do you need ROS?

Reply:
We need ROS because the `Unicycle2D` class is derived from the `fuse_core::AsyncMotionModel` which contains variables of `ros::NodeHandle` and `ros::AsyncSpinner` and if we don't initialize ROS, there is the following errors shown:
```
[FATAL ${node} ros.roscpp 1684306444.156373179]: You must call ros::init() before creating the first NodeHandle
[ERROR ${node} ros.roscpp 1684306444.167166713]: [registerPublisher] Failed to contact master at [:0].  Retrying...
```

Reply:
I see. Before the unit tests didn't create a `Unicycle2DTest` object, but now you create one.

I don't think there's any workaround to avoid this, so I think it's fine.